infrastructure/kzg/src/main/java/tech/pegasys/teku/kzg/ckzg4844/CKZG4844.java

```diff
@@ -0,0 +1,128 @@
+/*
+ * Copyright ConsenSys Software Inc., 2022
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package tech.pegasys.teku.kzg.ckzg4844;
+
+import ethereum.ckzg4844.CKzg4844JNI;
+import java.net.URL;
+import java.util.List;
+import java.util.stream.Stream;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.apache.tuweni.bytes.Bytes;
+import org.apache.tuweni.bytes.Bytes32;
+import tech.pegasys.teku.kzg.KZG;
+import tech.pegasys.teku.kzg.KZGCommitment;
+import tech.pegasys.teku.kzg.KZGException;
+import tech.pegasys.teku.kzg.KZGProof;
+
+/**
+ * Wrapper around jc-kzg-4844
+ *
+ * <p>This class should be a singleton
+ */
+public final class CKZG4844 implements KZG {
+
+  private static final Logger LOG = LogManager.getLogger();
+
+  private static CKZG4844 INSTANCE;
+
+  public static CKZG4844 createOrGetInstance() {
+    if (INSTANCE == null) {
+      INSTANCE = new CKZG4844();
+    }
+    return INSTANCE;
+  }
+
+  private CKZG4844() {
+    try {
+      LOG.info("Loaded C-KZG-4844 library");
+    } catch (final Exception ex) {
+      throw new KZGException("Failed to load C-KZG-4844 library", ex);
+    }
+  }
+
+  @Override
+  public void loadTrustedSetup(final URL trustedSetup) throws KZGException {
+    try {
+      final String file = CKZG4844Utils.copyTrustedSetupToTempFileIfNeeded(trustedSetup);
+      CKzg4844JNI.loadTrustedSetup(file);
+      LOG.info("Loaded trusted setup from {}", file);
```

Comment:
same here (debug), especially when it's tmp derived from resources it doesn't give any info to the user